src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/NewApiScannerInstaller.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * blackduck-common
+ *
+ * Copyright (c) 2024 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.blackduck.codelocation.signaturescanner.command;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+
+import com.synopsys.integration.blackduck.api.core.BlackDuckResponse;
+import com.synopsys.integration.blackduck.api.core.response.UrlSingleResponse;
+import com.synopsys.integration.blackduck.http.BlackDuckRequestBuilder;
+import com.synopsys.integration.blackduck.http.client.BlackDuckHttpClient;
+import com.synopsys.integration.blackduck.service.request.BlackDuckRequest;
+import com.synopsys.integration.blackduck.version.BlackDuckVersion;
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.io.FileUtils;
+
+import com.synopsys.integration.blackduck.exception.BlackDuckIntegrationException;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.log.IntLogger;
+import com.synopsys.integration.rest.HttpUrl;
+import com.synopsys.integration.rest.response.Response;
+import com.synopsys.integration.util.CleanupZipExpander;
+import com.synopsys.integration.util.OperatingSystemType;
+
+public class NewApiScannerInstaller extends ApiScannerInstaller {
```

Comment:
Comparing this class to OldAPiScannerInstaller would be the easiest way to review as they are both meant to have the same exact functionality

Reply:
Since new tends to become old, I would suggest not using those words as part of class names.
Perhaps something like `VersionedApiScannerInstaller` and `NonVersionedApiScannerInstaller` could work?

Reply:
That is a very good point, changed it to: 

NewApiScannerInstaller --> ToolsApiScannerInstaller (we call api/tools)
OldApiScannerInstaller --> ZipApiScannerInstaller (originally called the SignatureScannerZipInstaller... and the API path itself looks like /download/<operating system>.zip)